### ISSUE TITLE
As a user I want to be able to filter posts by role.

### ISSUE THREAD

**Issue by mathiasbrito-dci** [dci-python-backend-assignments/tdd-class-social#59]
As a user, I must be able to retrieve a list of posts filtered by the creator's role. e.g., by teacher, by student, by researcher, etc.

**Comment by Lamaosman2012** [comment]
@mathiasbrito-dci @divyaChandran10 
the role attribute has been deleted. Is the issue still relevant? we added a method to filter by creator_username, would it be alternative?

**Comment by Lamaosman2012** [comment]
@mathiasbrito-dci shall we add a `SearchView` or is it enough to do the search in the `localhosts/post/?search_terms=username` ?

**Comment by mathiasbrito-dci** [comment]
> @mathiasbrito-dci @divyaChandran10 the role attribute has been deleted. Is the issue still relevant? we added a method to filter by creator_username, would it be alternative?

we need to think about it, I think we can find a solution without the need of having the role, for example, thinking real quick here, if we want to limit the search by student we can just search on the student table.

**Comment by divyaChandran10** [comment]
I think, People in general will search any post by title regardless of the role.

**Comment by Lamaosman2012** [comment]
> > @mathiasbrito-dci @divyaChandran10 the role attribute has been deleted. Is the issue still relevant? we added a method to filter by creator_username, would it be alternative?
> 
> we need to think about it, I think we can find a solution without the need of having the role, for example, thinking real quick here, if we want to limit the search by student we can just search on the student table.

You mean we make new models for StudentPost, TeacherPost, InsitutionPost?

**Comment by Lamaosman2012** [comment]
> I think, People in general will search any post by title regardless of the role.

The Issue is about viewing Students/Teacher/Institution posts by just clicking on some filter button. or a link .

**Comment by divyaChandran10** [comment]
Yes I understood that, but since the role attribute is removed from models, can we do the common search of posts. If yes, we can change the issue as 'Search the posts by Title'.
